data/src/main/java/app/tivi/data/daos/EntityDao.kt

```diff
@@ -21,19 +21,34 @@ import androidx.room.Insert
 import androidx.room.Update
 import app.tivi.data.entities.TiviEntity
 
-interface EntityDao<in E : TiviEntity> {
+abstract class EntityDao<in E : TiviEntity> {
     @Insert
-    suspend fun insert(entity: E): Long
+    abstract suspend fun insert(entity: E): Long
 
     @Insert
-    suspend fun insertAll(vararg entity: E)
+    abstract suspend fun insertAll(vararg entity: E)
 
     @Insert
-    suspend fun insertAll(entities: List<E>)
+    abstract suspend fun insertAll(entities: List<E>)
 
     @Update
-    suspend fun update(entity: E)
+    abstract suspend fun update(entity: E)
 
     @Delete
-    suspend fun delete(entity: E): Int
+    abstract suspend fun deleteEntity(entity: E): Int
+
+    suspend fun insertOrUpdate(entity: E): Long {
```

Comment:
i think this is fixed in latest room such that you can have a concrete implementation in an interface. might be the upcoming patch release though.

Reply:
Cool, I'm on the latest but will migrate when it's out